signer/sign_state.go

```diff
@@ -138,33 +182,40 @@ func (signState *SignState) Save(
 	signState.Step = ssc.Step
 	signState.Signature = ssc.Signature
 	signState.SignBytes = ssc.SignBytes
+
+	jsonBytes, err := cometjson.MarshalIndent(signState, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+
+	signState.mu.Unlock()
+
+	signState.cond.Broadcast()
```

Comment:
Since `sync.Cond` is a rarely used part of the standard library, can you add a comment here explaining why/what we are broadcasting here?